### ISSUE TITLE
HTML is rendered

### ISSUE THREAD

**Issue by zsty** [qmlweb/qmlweb#407]
HTML is not escaped properly (if at all), causing code such as the following to have html left in
```qml
Text {
  text: "Hello <h1>world</h1>!"
}
```
This isn't much of an issue, but has a really easy fix. I've put one below, it should probably work (I can't think of a case where it wouldn't)
```js
text.replace(/&/g, "&amp;")
    .replace(/</g, "&lt;");
```

**Comment by ChALkeR** [comment]
Thanks for bringing this up.

That is the expected default behaviour per the Qt QML docs, see [Text#text](http://doc.qt.io/qt-5/qml-qtquick-text.html#text-prop) and [Text#textFormat](http://doc.qt.io/qt-5/qml-qtquick-text.html#textFormat-prop).

You could also reproduce that behaviour with Qt QML on your example.

In QmlWeb, changing the `textFormat` is not supported yet, but is planned and should be fairly easy to implement.

That said, the fix would look different, as QmlWeb does not generate the HTML, it instead produces the DOM tree, and to render something as text it is sufficient to create a text node using `Document.createTextNode()`.

Another way of improving things on QmlWeb side would be to filter the allowed content in the rich text inside the `text` property — as Qt supports only a limited subset of tags there (that's not really HTML, just some tags for formatting: http://doc.qt.io/qt-5/richtext-html-subset.html), and either using a thirdparty sanitizer with an appropriate ruleset or setting the content also through generating safe DOM instead of directly assigning `innerHtml`.

**Comment by ChALkeR** [comment]
@zsty I believe this is mostly solved in d439145c1d7ed2c0923508d9a73ed5f9887467ae.

To opt-out from rendering HTML, you can now use `textFormat: Text.PlainText` (like in Qt QML).

The default is Text.AutoText — if there are no tags found in the text, it's rendered as PlainText, and if there are tags found, its rendered as HTML subset (sanitization not implemented for now).

E.g., with `textFormat: Text.AutoText` (default):
 * "Hello&amp;!" renders as "Hello&amp;!" (plain)
 * "Hello&amp;<b>xx</b>!" renders as "Hello&xx!" (styled)

With `textFormat: Text.PlainText` everything is rendered as plain text (like if it was escaped).

I believe this issue can be closed now as resolved. I will open a separate issue for sanitization of `Text.StyledText` and `Text.RichText`.
